### PR TITLE
updates VANotify::InProgressFormReminder

### PR DESCRIPTION
**Pull request by MarchandMD**
## Summary

- *This work is behind a feature toggle (flipper): YES/NO* 

> Yes; It's behind `:va_notify_user_account_job` which is currently on in Staging only

- *(Summarize the changes that have been made to the platform)*

> before this PR: 
> was passing `veteran.uuid` to `UserAccountJob` within `VANotify::InProgressFormReminder` job
> 
> after this PR:
> now passing `in_progress_form.user_account_id`

- *(If bug, how to reproduce)* 

> N/A

- *(What is the solution, why is this the solution?)*

> While testing UserAccountJob in Staging, determined that the wrong value was being passed. this is the solution because the UserAccount will now be able to be found

- *(Which team do you work for, does your team own the maintenance of this component?)*

> VANotify

- *(If introducing a flipper, what is the success criteria being targeted?)*

> N/A

## Related issue(s)

- *Link to ticket created in va.gov-team repo OR screenshot of Jira ticket if your team uses Jira*
- *Link to previous change of the code/bug (if applicable)*
- *Link to epic if not included in ticket*

## Testing done

- [ ] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by nathanbwright — APPROVED** [review]

File: modules/va_notify/spec/services/in_progress_form_reminder_spec.rb [hunk 1]
```diff
@@ -72,19 +74,25 @@
...
-                                                                                 'form_age' => ''
-                                                                               })
+        # rubocop:disable Layout:LineLength
+        expect(VANotify::UserAccountJob).to have_received(:perform_async).with(in_progress_form.user_account_id, 'fake_template_id',
+                                           {
+                                             'first_name' => 'FIRST_NAME',
+                                             'date' => expiration_date,
+                                             'form_age' => ''
+                                           })
+        # rubocop:enable Layout:LineLength
```
Comment (rmtolmach) [inline modules/va_notify/spec/services/in_progress_form_reminder_spec.rb:84]: Is it possible to get this working without disabling the line length? What if you broke the line at `.with(`?

File: modules/va_notify/spec/sidekiq/in_progress_form_reminder_spec.rb [hunk 2]
```diff
@@ -42,25 +45,28 @@
...
 
-        expect(VANotify::UserAccountJob).to have_received(:perform_async).with('uuid', 'fake_template_id',
+        # rubocop:disable Layout/LineLength
+        expect(VANotify::UserAccountJob).to have_received(:perform_async).with(in_progress_form_with_user_account_id.user_account_id, 'fake_template_id',
                                                                                {
                                                                                  'first_name' => 'FIRST_NAME',
                                                                                  'date' => expiration_date,
                                                                                  'form_age' => ''
                                                                                })
+        # rubocop:enable Layout/LineLength
```
Comment (rmtolmach) [inline modules/va_notify/spec/sidekiq/in_progress_form_reminder_spec.rb:61]: here, too?

File: modules/va_notify/spec/sidekiq/in_progress_form_reminder_spec.rb [hunk 3]
```diff
@@ -42,25 +45,28 @@
...
       end
     end
 
     describe 'multiple relevant in_progress_forms' do
       let!(:in_progress_form_1) do
         Timecop.freeze(7.days.ago)
-        in_progress_form = create(:in_progress_686c_form, user_uuid: user.uuid)
+        in_progress_form = create(:in_progress_686c_form, user_uuid: user.uuid,
+                                                          user_account_id: create(:user_account).id)
```
Comment (rmtolmach) [inline modules/va_notify/spec/sidekiq/in_progress_form_reminder_spec.rb:69]: non-blocking, but while you're updating formatting, what do you think of this instead? 
```suggestion
        in_progress_form = create(
          :in_progress_686c_form, user_uuid: user.uuid, user_account_id: create(:user_account).id
        )
```

File: modules/va_notify/spec/services/in_progress_form_reminder_spec.rb (hunk 1 above)
Comment (MarchandMD) [reply modules/va_notify/spec/services/in_progress_form_reminder_spec.rb]: Yes it is! Thank you!

File: modules/va_notify/spec/sidekiq/in_progress_form_reminder_spec.rb (hunk 3 above)
Comment (MarchandMD) [reply modules/va_notify/spec/sidekiq/in_progress_form_reminder_spec.rb]: I like it well enough. Applied!

**Pull request merged**